pageserver/src/tenant/disk_btree.rs

```diff
@@ -259,27 +260,32 @@ where
         Ok(result)
     }
 
-    pub fn iter<'a>(
-        &'a self,
-        start_key: &'a [u8; L],
-        ctx: &'a RequestContext,
-    ) -> DiskBtreeIterator<'a> {
+    pub fn iter<'a>(self, start_key: &'a [u8; L], ctx: &'a RequestContext) -> DiskBtreeIterator<'a>
+    where
+        R: 'a,
+    {
         DiskBtreeIterator {
-            stream: Box::pin(self.get_stream_from(start_key, ctx)),
+            stream: Box::pin(self.into_stream(start_key, ctx)),
         }
     }
 
     /// Return a stream which yields all key, value pairs from the index
     /// starting from the first key greater or equal to `start_key`.
     ///
-    /// Note that this is a copy of [`Self::visit`].
+    /// Note 1: that this is a copy of [`Self::visit`].
     /// TODO: Once the sequential read path is removed this will become
     /// the only index traversal method.
-    pub fn get_stream_from<'a>(
-        &'a self,
+    ///
+    /// Note 2: this function used to take `&self` but due to the iterator-based API refactor,
+    /// it now consumes `self`. Feel free to add the `&self` variant back if it's necessary.
+    pub fn into_stream<'a>(
```

Comment:
I assume `iterator-based API refactor` refers to the latest changes in this PR?
Nobody will know what that means down the road.
Better to point out why the bottommost compaction needs to consume `self` and can't
deal with `&self`.